etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,220 @@
+                SET outcome_date = CURDATE() WHERE outcome_date > CURDATE();
+        ''')
+
+        // TODO: investigate cases where art_start_date > outcome_date
+
+        executeMysql("Remove bad health center entries", '''
+            UPDATE hivmigration_programs_raw
+            SET health_center = IF (health_center = 0 OR health_center = 42, NULL, health_center),
+                starting_health_center = IF (starting_health_center = 0 OR starting_health_center = 42, NULL, starting_health_center)
+            ;
+        ''')
+
+        executeMysql("Ensure health_center is set and drop superfluous values for starting_health_center", '''
+            UPDATE hivmigration_programs_raw
+            SET
+                health_center = COALESCE(health_center, starting_health_center, 1),
+                starting_health_center = IF (
+                        starting_health_center = health_center OR
+                        health_center_transfer_date IS NULL OR
+                        health_center_transfer_date > outcome_date OR
+                        health_center_transfer_date < enrollment_date,
+                    NULL,
+                    starting_health_center)
+            ;
+        ''')
+
+        executeMysql("Load single enrollments to hivmigration_programs table", '''
+            INSERT INTO hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT source_patient_id, health_center, enrollment_date, art_start_date, outcome_date, outcome FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NULL
+            ;
+        ''')
+
+        executeMysql("Load initial enrollments into hivmigration_programs table", '''
+            SET @outcome_transferred_out = (select concept_id from concept where uuid = '3cdd5c02-26fe-102b-80cb-0017a47871b2');
+
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                starting_health_center,
+                enrollment_date,
+                IF (art_start_date < health_center_transfer_date, art_start_date, NULL),
+                health_center_transfer_date,
+                @outcome_transferred_out
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load current enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                health_center,
+                health_center_transfer_date,
+                IF (art_start_date < health_center_transfer_date, health_center_transfer_date, art_start_date),
```

Comment:
I don't think this is right.  Why are you checking art_start_date vs. health_center_transfer_date?  The transfer date is the transfer date.  It's certainly possible they had an enrollment with no ART treatment at their initial site.

Reply:
Okay, I'm changing it to just leave art_start_date alone. We can figure out how we should adjust it once we start using art_start_date for anything (presumably once we add the patient_state logic).

Not sure about the comment "It's certainly possible they had an enrollment"... if `art_start_date` is null then the inequality returns false and the expression returns `art_start_date` which is null.